### PR TITLE
2627 Enhance surface metrics warning

### LINKED ISSUE #2627: SurfaceDistanceMetric (SDM) and HausdorffDistanceMetric (HDM) print wrong all-zero warning messages
**Describe the bug**
SDM and HDM incorrectly warn that both prediction and ground truth are all 0 (for some class) even when only one of them is all 0 (for some class):

**To Reproduce**
Steps to reproduce the behavior (for HDM):

```
import torch
import monai
metric = monai.metrics.HausdorffDistanceMetric()
#forward direction
metric(torch.cat((torch.ones(1,1,256,256),torch.zeros(1,1,256,256)), dim=1), torch.randint(0,2,(1,2,256,256)))

/autofs/space/vault_021/users/siy0/anaconda3/lib/python3.8/site-packages/monai/metrics/utils.py:199: UserWarning: prediction is all 0, this may result in nan/inf distance.
  warnings.warn("prediction is all 0, this may result in nan/inf distance.")
/autofs/space/vault_021/users/siy0/anaconda3/lib/python3.8/site-packages/monai/metrics/utils.py:195: UserWarning: ground truth is all 0, this may result in nan/inf distance.
  warnings.warn("ground truth is all 0, this may result in nan/inf distance.")

#backward direction
metric(torch.randint(0,2,(1,2,256,256)),torch.cat((torch.ones(1,1,256,256),torch.zeros(1,1,256,256)), dim=1))

/autofs/space/vault_021/users/siy0/anaconda3/lib/python3.8/site-packages/monai/metrics/utils.py:195: UserWarning: ground truth is all 0, this may result in nan/inf distance.
  warnings.warn("ground truth is all 0, this may result in nan/inf distance.")
/autofs/space/vault_021/users/siy0/anaconda3/lib/python3.8/site-packages/monai/metrics/utils.py:199: UserWarning: prediction is all 0, this may result in nan/inf distance.
  warnings.warn("prediction is all 0, this may result in nan/inf distance.")

```

The same bug is there if SDM is run with `symmetric=True`

**Expected behavior**
Only the first warning should be displayed in each of the two above cases.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2627 .

### Description
This PR enhanced the confusing warning messages of distance metrics.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by yiheng-wang-nv — APPROVED** [review]
Thanks, I think this change is better than the original implementation to avoid "duplicated" warnings when `directed == False` for `HausdorffDistanceMetric`

**Comment by Nic-Ma** [comment]
Hi @yiheng-wang-nv ,

Thanks for your review.
The main problem is when exchanged pred and label to compute distance, the warning message is wrong.

Thanks.
